diff --git a/desktop/source/app/app.cxx b/desktop/source/app/app.cxx
index 1643aaf4e..37cfd0f4e 100644
--- a/desktop/source/app/app.cxx
+++ b/desktop/source/app/app.cxx
@@ -1268,8 +1268,6 @@ void Desktop::Exception(ExceptionCategory nCategory)
     if( bRestart )
     {
         RequestHandler::Disable();
-        if( pSignalHandler )
-            osl_removeSignalHandler( pSignalHandler );
 
         restartOnMac(false);
 #if !ENABLE_WASM_STRIP_SPLASH
@@ -1277,7 +1275,7 @@ void Desktop::Exception(ExceptionCategory nCategory)
             m_rSplashScreen->reset();
 #endif
 
-        _exit( EXITHELPER_CRASH_WITH_RESTART );
+        return;
     }
     else
     {
