diff --git a/desktop/source/app/app.cxx b/desktop/source/app/app.cxx
index da313a19a..32bcb8c01 100644
--- a/desktop/source/app/app.cxx
+++ b/desktop/source/app/app.cxx
@@ -1226,8 +1226,6 @@ void Desktop::Exception(ExceptionCategory nCategory)
     if( bRestart )
     {
         RequestHandler::Disable();
-        if( pSignalHandler )
-            osl_removeSignalHandler( pSignalHandler );
 
         restartOnMac(false);
 #if !ENABLE_WASM_STRIP_SPLASH
@@ -1235,7 +1233,7 @@ void Desktop::Exception(ExceptionCategory nCategory)
             m_rSplashScreen->reset();
 #endif
 
-        _exit( EXITHELPER_CRASH_WITH_RESTART );
+        return;
     }
     else
     {
