diff --git a/include/sal/config.h b/include/sal/config.h
index f666fed23..1e8fe4674 100644
--- a/include/sal/config.h
+++ b/include/sal/config.h
@@ -48,7 +48,7 @@
 #endif /* defined _WIN32 */
 
 #if defined(__sun) || defined(LINUX) || defined(NETBSD) || defined(FREEBSD) || defined(OPENBSD)    \
-    || defined(DRAGONFLY) || defined(ANDROID) || defined(HAIKU)
+    || defined(DRAGONFLY) || defined(__ANDROID__) || defined(HAIKU)
 #define SAL_UNX
 #define SAL_DLLEXTENSION ".so"
 #define SAL_EXEEXTENSION ""
