diff --git a/vcl/unx/gtk3/gtkdata.cxx b/vcl/unx/gtk3/gtkdata.cxx
index 5f9ed40fc..1f87aa6f8 100644
--- a/vcl/unx/gtk3/gtkdata.cxx
+++ b/vcl/unx/gtk3/gtkdata.cxx
@@ -752,7 +752,20 @@ extern "C" {
 
         ImplSVData* pSVData = ImplGetSVData();
         if( pSVData->maSchedCtx.mpSalTimer )
+        {
+#if defined(__ANDROID__)
+            try
+            {
+                pSVData->maSchedCtx.mpSalTimer->CallCallback();
+            }
+            catch (...)
+            {
+                GetGtkSalData()->setException(std::current_exception());
+            }
+#else
             pSVData->maSchedCtx.mpSalTimer->CallCallback();
+#endif
+        }
 
         return G_SOURCE_REMOVE;
     }
@@ -848,7 +861,18 @@ extern "C" {
         {
             GtkSalDisplay *pThisDisplay = static_cast<GtkSalData *>(data)->GetGtkDisplay();
             assert(static_cast<const SalGenericDisplay *>(pThisDisplay) == pDisplay);
+#if defined(__ANDROID__)
+            try
+            {
+                pThisDisplay->DispatchInternalEvent();
+            }
+            catch (...)
+            {
+                static_cast<GtkSalData *>(data)->setException(std::current_exception());
+            }
+#else
             pThisDisplay->DispatchInternalEvent();
+#endif
         }
         return true;
     }
