diff --git a/configure.ac b/configure.ac
index 7b5c30c3b..1038ab152 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4236,7 +4236,7 @@ if test -n "$WIN_MULTI_ARCH" -a -n "$WIN_OTHER_ARCH"; then
 fi
 
 
-if test "$build_cpu" != "$host_cpu" -o "$DISABLE_DYNLOADING" = TRUE; then
+if test "$build_cpu" != "$host_cpu"; then
     # To allow building Windows multi-arch releases without cross-tooling
     if test "$DISABLE_DYNLOADING" = TRUE -o \( -z "$WIN_MULTI_ARCH" -a -z "$WIN_OTHER_ARCH" \); then
         cross_compiling="yes"
